wgengine/netstack/netstack.go

```diff
@@ -859,7 +861,14 @@ func (ns *Impl) acceptTCP(r *tcp.ForwarderRequest) {
 
 	if ns.lb != nil {
 		if reqDetails.LocalPort == 22 && ns.processSSH() && ns.isLocalIP(dialIP) {
-			c := createConn()
+			// Use a higher keepalive idle time for SSH connections, as they are
+			// typically long lived and idle connections are more likely to be
+			// intentional. Ideally we would turn this off entirely, but we can't
+			// tell the difference between a long lived connection that is idle
+			// vs a connection that is dead because the peer has gone away.
```

Comment:
I imagine our long-term fix is to get actual packet forwarding into netstack, rather than terminate the TCP session on one side and made a new one other the other side.

Reply:
Yeah, but I think we will have something like this for Tailscale SSH even then as we have to handle the incoming connection in process.